src/analyze/analyze-plot.c

```diff
@@ -178,40 +179,26 @@ static int plot_unit_times(UnitTimes *u, double width, int y) {
         return 1;
 }
 
-int verb_plot(int argc, char *argv[], void *userdata) {
-        _cleanup_(free_host_infop) HostInfo *host = NULL;
-        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
-        _cleanup_(unit_times_free_arrayp) UnitTimes *times = NULL;
-        _cleanup_free_ char *pretty_times = NULL;
-        bool use_full_bus = arg_scope == LOOKUP_SCOPE_SYSTEM;
-        BootTimes *boot;
-        UnitTimes *u;
-        int n, m = 1, y = 0, r;
-        double width;
-
-        r = acquire_bus(&bus, &use_full_bus);
-        if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
-
-        n = acquire_boot_times(bus, &boot);
-        if (n < 0)
-                return n;
-
-        n = pretty_boot_time(bus, &pretty_times);
-        if (n < 0)
-                return n;
-
-        if (use_full_bus || arg_scope != LOOKUP_SCOPE_SYSTEM) {
-                n = acquire_host_info(bus, &host);
-                if (n < 0)
-                        return n;
-        }
-
-        n = acquire_time_data(bus, &times);
-        if (n <= 0)
-                return n;
+static void limit_times_to_boot(const BootTimes* boot, UnitTimes* u){
+        if (u->deactivated > u->activating &&
+            u->deactivated <= boot->finish_time &&
+            u->activated == 0 && u->deactivating == 0)
+                u->activated = u->deactivating = u->deactivated;
+        if (u->activated < u->activating || u->activated > boot->finish_time)
+                u->activated = boot->finish_time;
+        if (u->deactivating < u->activated || u->deactivating > boot->finish_time)
+                u->deactivating = boot->finish_time;
+        if (u->deactivated < u->deactivating || u->deactivated > boot->finish_time)
+                u->deactivated = boot->finish_time;
+}
 
-        typesafe_qsort(times, n, compare_unit_start);
+static int produce_plot_as_svg(const UnitTimes* times,
+                               const HostInfo* host,
+                               const BootTimes* boot,
+                               const char* pretty_times) {
```

Comment:
we usually break long function parameter lists like so:

```c
static int produce_plot_as_svg(
        const UnitTimes* …,
        …,
        …) {
```

i.e. already break before the first parameter, and indent to double the usual indentation (i.e. 16 spaces).

some older code doesn't do that yet, but let's do that for all new code.